File: kinotic-js/workspace/packages/vm-manager/src/model/VmNodeRegistration.ts
/**
 * DTO sent by a vm-manager process when it registers with the server.
 * Contains only the information the node knows about itself.
 */
export class VmNodeRegistration {

    /**
     * Unique identifier for this node.
     */
    public id: string

    /**
     * Human-readable name for the node.
     */
    public name: string

    /**
     * The hostname or address of the node.
     */
    public hostname: string

    /**
     * Total number of vCPUs available on this node.
     */
    public totalCpus: number = 0

    /**
     * Total memory available on this node in megabytes.
     */
    public totalMemoryMb: number = 0

    /**
     * Total disk space available on this node in megabytes.
     */
    public totalDiskMb: number = 0

    constructor(id: string, name: string, hostname: string) {
        this.id = id
        this.name = name
        this.hostname = hostname
    }
}

File: kinotic-system-api/src/main/java/org/kinotic/system/api/workload/VmNodeRegistration.java
package org.kinotic.system.api.workload;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;
import org.kinotic.management.api.model.workload.VmProviderType;

/**
 * DTO sent by a vm-manager process when it registers with the server.
 * Contains only the information the node knows about itself.
 */
@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
public class VmNodeRegistration {

    /**
     * Unique identifier for this node.
     */
    private String id;

    /**
     * Human-readable name for the node.
     */
    private String name;

    /**
     * The hostname or address of the node.
     */
    private String hostname;

    /**
     * The VM provider this node runs every workload on.
     */
    private VmProviderType providerType = VmProviderType.BOXLITE;

    /**
     * Total number of vCPUs available on this node.
     */
    private int totalCpus;

    /**
     * Total memory available on this node in megabytes.
     */
    private int totalMemoryMb;

    /**
     * Total disk space available on this node in megabytes.
     */
    private int totalDiskMb;

    /**
     * Base directory every workload volume mount on this node must live under.
     */
    private String workloadDataDir;

    public VmNodeRegistration(String id, String name, String hostname) {
        this.id = id;
        this.name = name;
        this.hostname = hostname;
    }
}
